actix-web v4.2.0's actix-http dependency is not specific enough

## Expected Behavior
`actix-web` v4.2.0 successfully build, as long as its transitive dependencies all satisfy the version constraints.

## Current Behavior
`actix-web` v4.2.0 doesn't build with actix-http v3.2.0 despite the actix-http version satisfies the constraint.

https://github.com/actix/actix-web/blob/7767cf30710ec2a8666b0c9e6f578aa590038a54/actix-web/Cargo.toml#L74

## Possible Solution
Change `Cargo.toml` in `actix-web` to point to `actix-http` 3.2.2+.

## Steps to Reproduce (for bugs)
Create a repo with `actix-web` pinned to `v4.2.0` and `actix-http` `v3.2.0`.
`cargo build` fails with the following.

```console
   Compiling actix-web v4.2.0
error[E0277]: the trait bound `actix_http::header::ExtendedValue: std::cmp::Eq` is not satisfied
   --> $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-web-4.2.0/src/http/header/content_disposition.rs:99:17
    |
82  | #[derive(Clone, Debug, PartialEq, Eq)]
    |                                   -- in this derive macro expansion
...
99  |     FilenameExt(ExtendedValue),
    |                 ^^^^^^^^^^^^^ the trait `std::cmp::Eq` is not implemented for `actix_http::header::ExtendedValue`
    |
note: required by a bound in `AssertParamIsEq`
   --> $HOME/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/cmp.rs:317:31
    |
317 | pub struct AssertParamIsEq<T: Eq + ?Sized> {
    |                               ^^ required by this bound in `AssertParamIsEq`
    = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)
```

## Context
The repo has `Cargo.lock`. Dependabot bumps the actix-web, which causes the build to fail because `actix-http` is not new enough.

## Your Environment
- Rust Version (I.e, output of `rustc -V`): `rustc 1.63.0 (4b91a6ea7 2022-08-08)`
- Actix Web Version: 4.2.0

--- a/actix-web/CHANGES.md
+++ b/actix-web/CHANGES.md
@@ -1,6 +1,10 @@
 # Changelog
 
 ## Unreleased - 2022-xx-xx
+### Fixed
+- Bump minimum version of `actix-http` dependency to fix compatibility issue. [#2871]
+
+[#2871]: https://github.com/actix/actix-web/pull/2871
 
 
 ## 4.2.0 - 2022-09-11

--- a/actix-web/Cargo.toml
+++ b/actix-web/Cargo.toml
@@ -71,7 +71,7 @@ actix-service = "2"
 actix-utils = "3"
 actix-tls = { version = "3", default-features = false, optional = true }
 
-actix-http = { version = "3", features = ["http2", "ws"] }
+actix-http = { version = "3.2.2", features = ["http2", "ws"] }
 actix-router = "0.5"
 actix-web-codegen = { version = "4.1", optional = true }
 

